src/testrender/testrender.cpp

```diff
@@ -368,7 +368,7 @@ main(int argc, const char* argv[])
 
     // We're done with the shading system now, destroy it
     rend->clear();
-    delete rend;
     delete shadingsys;
+    delete rend;
```

Comment:
The sanitizer CI tests are failing for this PR, and I wonder if it's related to this deletion reversal here?

Reply:
When the renderer is being destroyed it calls back into the shading system, deleting the shading system before the renderer is definitely erroneous. I don't see anything in the CI output that sheds much light on the failure.

Reply:
Hmmm, I'll try to repro on my end.

Reply:
Ah, I just managed to get the full log and I can see the errors reported in the `ShadingSystem` destructor. I can drop this change from the PR and then I can investigate these issues in isolation.